alicloud/service_alicloud_cas.go

```diff
@@ -0,0 +1,40 @@
+package alicloud
+
+import (
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/cas"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+type CasService struct {
+	client *connectivity.AliyunClient
+}
+
+func (s *CasService) DescribeCas(id string) (*cas.Certificate, error) {
+	var cert *cas.Certificate
+
+	args := cas.CreateDescribeUserCertificateListRequest()
+	args.ShowSize = requests.NewInteger(50)
+	args.CurrentPage = requests.NewInteger(1)
+
+	raw, err := s.client.WithCasClient(func(casClient *cas.Client) (interface{}, error) {
+		return casClient.DescribeUserCertificateList(args)
+	})
+	if err != nil {
+		return nil, err
+	}
+	res, _ := raw.(*cas.DescribeUserCertificateListResponse)
+
+	if len(res.CertificateList) == 0 {
+		return nil, nil
```

Comment:
If there is no matching certificate, please return a NotFoundError

Reply:
done